src/benchmark.rs

```diff
@@ -5,8 +5,14 @@ pub fn run(count: usize, size: usize) {
     let buffers = generate_buffers(count, size);
     let k = keccak_bench(&buffers);
 
-    let signatures = generate_signatures(buffers);
-    ecrecover_bench(signatures, k);
+    let signatures = generate_signatures(&buffers);
```

Comment:
It would make sense to generate signatures once and pass the reference to benchmark functions.